src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,692 @@
+            "close_ksize", fallback_value=11
+        )
+
+        self.input_plugin = self._extract_dict_parameter("oav", fallback_value="OAV")
+        self.mxsc_input = self._extract_dict_parameter(
+            "mxsc_input", fallback_value="CAM"
+        )
+        self.min_callback_time = self._extract_dict_parameter(
+            "min_callback_time", fallback_value=0.08
+        )
+
+        self.direction = self._extract_dict_parameter("direction")
+
+        self.max_tip_distance = self._extract_dict_parameter("max_tip_distance")
+
+    def _extract_dict_parameter(self, key: str, fallback_value=None, reload_json=False):
+        """
+        Designed to extract parameters from the json OAVParameters.json. This will hopefully be changed in
+        future, but currently we have to use the json passed in from GDA
+
+        The json is of the form:
+            {
+                "parameter1": value1,
+                "parameter2": value2,
+                "context_name": {
+                    "parameter1": value3
+            }
+        When we extract the parameters we want to check if the given context (stored as a class variable)
+        contains a parameter, if it does we return it, if not we return the global value. If a parameter
+        is not found at all then the passed in fallback_value is returned. If that isn't found then an
+        error is raised.
+
+        Args:
+            key: the key of the value being extracted
+            fallback_value: a value to be returned if the key is not found
+            reload_json: reload the json from the file before searching for it, needed because some
+                parameters can change mid operation.
+
+        Returns: The extracted value corresponding to the key, or the fallback_value if none is found.
+        """
+
+        if reload_json:
+            self.load_json()
+
+        if self.context in self.parameters:
+            if key in self.parameters[self.context]:
+                return self.parameters[self.context][key]
+
+        if key in self.parameters:
+            return self.parameters[key]
+
+        if fallback_value:
+            return fallback_value
+
+        # No fallback_value was given and the key wasn't found
+        raise KeyError(
+            f"Searched in {self.file} for key {key} in context {self.context} but no value was found. No fallback value was given."
+        )
+
+    def load_microns_per_pixel(self, zoom):
+        zoom_levels_filename = "src/artemis/oav/microns_for_zoom_levels.json"
+        f = open(zoom_levels_filename)
+        json_dict = json.load(f)
+        self.micronsPerXPixel = json_dict[zoom]["micronsPerXPixel"]
+        self.micronsPerYPixel = json_dict[zoom]["micronsPerYPixel"]
+        if self.micronsPerXPixel is None or self.micronsPerYPixel is None:
+            raise KeyError(
+                f"Could not find the micronsPer[X,Y]Pixel parameters in {zoom_levels_filename}."
+            )
+
+
+class OAVCentring:
```

Comment:
Should: I'm not sure it makes sense to have a class for this. Usually plans are just generators that you pass in all the devices for